packages/catalog-process/test/db.test.ts

```diff
@@ -68,14 +117,1078 @@ describe("database test", async () => {
+      it("should write on event-store for the deletion of an eService", async () => {
+        await addOneEService(mockEService);
+        await catalogService.deleteEService(
+          mockEService.id,
+          getMockAuthData(mockEService.producerId)
+        );
+        const writtenEvent = await readLastEventByStreamId(mockEService.id);
+        expect(writtenEvent.stream_id).toBe(mockEService.id);
+        expect(writtenEvent.version).toBe("1");
+        expect(writtenEvent.type).toBe("EServiceDeleted");
+        const writtenPayload = decode({
+          messageType: EServiceDeletedV1,
+          payload: writtenEvent.data,
+        });
+        expect(writtenPayload.eServiceId).toBe(mockEService.id);
+      });
+
+      it("should throw eServiceNotFound if the eService doesn't exist", () => {
+        expect(
+          catalogService.deleteEService(
+            mockEService.id,
+            getMockAuthData(mockEService.producerId)
+          )
+        ).rejects.toThrowError(eServiceNotFound(mockEService.id));
+      });
+
+      it("should throw operationForbidden if the requester is not allowed", async () => {
+        await addOneEService(mockEService);
+        expect(
+          catalogService.deleteEService(mockEService.id, getMockAuthData())
+        ).rejects.toThrowError(operationForbidden);
+      });
+
+      it("should throw eServiceCannotBeDeleted if the eService has a descriptor", async () => {
+        const descriptor: Descriptor = {
+          ...mockDescriptor,
+          state: descriptorState.published,
+        };
+        const eService: EService = {
+          ...mockEService,
+          descriptors: [descriptor],
+        };
+        await addOneEService(eService);
+        expect(
+          catalogService.deleteEService(
+            eService.id,
+            getMockAuthData(eService.producerId)
+          )
+        ).rejects.toThrowError(eServiceCannotBeDeleted(eService.id));
+      });
+    });
+
+    describe("create descriptor", async () => {
+      it("should write on event-store for the creation of a descriptor", async () => {
+        await addOneEService(mockEService);
+        await catalogService.createDescriptor(
+          mockEService.id,
+          buildDescriptorSeed(mockDescriptor),
+          getMockAuthData(mockEService.producerId)
+        );
+        const writtenEvent = await readLastEventByStreamId(mockEService.id);
+        expect(writtenEvent.stream_id).toBe(mockEService.id);
+        expect(writtenEvent.version).toBe("1");
+        expect(writtenEvent.type).toBe("EServiceDescriptorAdded");
+        const writtenPayload = decode({
+          messageType: EServiceDescriptorAddedV1,
+          payload: writtenEvent.data,
+        });
+
+        const descriptor: Descriptor = {
+          ...mockDescriptor,
+          createdAt: new Date(
+            Number(writtenPayload.eServiceDescriptor?.createdAt)
+          ),
+          id: writtenPayload.eServiceDescriptor?.id,
+        };
```

Comment:
```suggestion
        expect(writtenPayload.eServiceDescriptor).toBeDefined();

        const descriptor: Descriptor = {
          ...mockDescriptor,
          createdAt: new Date(
            Number(writtenPayload.eServiceDescriptor!.createdAt)
          ),
          id: writtenPayload.eServiceDescriptor!.id,
        };
```

Reply:
Updated only in the line that raised a typing error.